QGA-9 Create quizzes routes

sadinovsr opened the pull request.

Commit:
REFERENCE: QGA-9
  DESCRIPTION: Added quiz route to main server file, created quiz router
    and created quizController with method which gets all quizzes.

Commit:
REFERENCE: QGA-9
  DESCRIPTION: Added addQuiz, updateQuiz and deleteQuiz to
    quizController

Commit:
REFERENCE: QGA-9
  DESCRIPTION: Created additional check whether quiz exists in database,
    before trying to delete it.

Gruseckis commented:
You will need to resolve merge conflicts before merging

Gruseckis commented on controllers/quizController.js:
```diff
@@ -0,0 +1,59 @@
+  try {
+    const quizzes = await QuizModel.getAllQuizzes();
+    res.status(200).send({
+      payload: { quizzes }
```
please return empty array if no quizzes were found

sadinovsr replied:
model.find() already returns empty array by default if nothing found.

Gruseckis replied:
Ok. Did not know that

sadinovsr committed:
Merge branch 'development' into QGA-9-Create-quizzes-routes

Gruseckis commented on controllers/quizController.js:
```diff
@@ -0,0 +1,59 @@
+    const quiz = await QuizModel.save({
+      ownerId: req.body.userId,
+      name: req.body.name,
+      description: req.body.description
```
Questions are missing, but we need to check how quiz saving will be handled by front end

Gruseckis commented on controllers/quizController.js:
```diff
@@ -0,0 +1,59 @@
+    if (!quiz) {
+      throw new AppError('Cannot delete quiz which does not exist');
+    }
+    await QuizModel.deleteQuizById(req.params.quizId);
```
delete quiz by id should return deleted quiz so previous check is redundant.

Gruseckis commented on server.js:
```diff
@@ -5,6 +5,7 @@ import defaultErrorHandler from './middlewares/defaultErrorHandler';
 import mongoose from 'mongoose';
 import bodyParser from 'body-parser';
 import index from './routes/index';
+import quiz from './routes/quiz';
```
Please give it more meaningful name like quizRoutes

Commit:
REFERENCE: QGA-9
  DESCRIPTION: Changed existing quiz validation so we do not create two
    different database calls

Commit:
Merge branch 'QGA-9-Create-quizzes-routes' of https://github.com/Gruseckis/quiz-game-api into QGA-9-Create-quizzes-routes

Commit:
REFERENCE: QGA-9
  DESCRIPTION: Added more meaningful name to quiz routes in server.js

Commit:
REFERENCE: QGA-9
  DESCRIPTION: Changed router named quiz.js to quizRouter.js so it is
    more specific and understandable.

sadinovsr committed:
Merge branch 'development' into QGA-9-Create-quizzes-routes
